gpytorch/lazy/kronecker_product_added_diag_lazy_tensor.py

```diff
@@ -46,8 +46,8 @@ def inv_quad_logdet(self, inv_quad_rhs=None, logdet=False, reduce_inv_quad=True)
         return inv_quad_term, logdet_term
 
     def _logdet(self):
-        eig_matrix, _ = self.lazy_tensor._symeig()
-        evals_plus_diag = eig_matrix.diag() + self.diag_tensor.diag()
+        evals, _ = self.lazy_tensor.symeig(eigenvectors=False)
+        evals_plus_diag = evals + self.diag_tensor.diag()
```

Comment:
Might as well check that the diagonal tensor is (1) constant and (2) positive here as well -- I don't think `add` checks for that, so you could technically end up with this LT not meeting those conditions right?

Reply:
This one is really on me; as it is, this is really only KP + ConstantDiag throughout. I need to make a PR for the general Kronecker structured case.